.github/workflows/ci.yml

```diff
@@ -36,6 +37,9 @@ jobs:
         git clone https://github.com/capnproto/capnproto-java.git $GITHUB_WORKSPACE/env/capnproto-java
         git clone https://github.com/SymbiFlow/fpga-interchange-schema.git $GITHUB_WORKSPACE/env/fpga-interchange-schema
 
+    - name: Test testarch
+      run: ./.github/workflows/testarch_device.sh
+
```

Comment:
Before calling this it is good to use the [ccache](https://github.com/hendrikmuhs/ccache-action) action to speed up capnp compilation between different CI builds.

To do so you can [this](https://github.com/YosysHQ/nextpnr/blob/ded32f33906e2547eec68d91715c0759290c6a1f/.github/workflows/interchange_ci.yml#L21-L22) before `Test testarch`, and export the [ccache path](https://github.com/YosysHQ/nextpnr/blob/ded32f33906e2547eec68d91715c0759290c6a1f/.github/workflows/interchange_ci.yml#L39)